spec/ddtrace/contrib/concurrent_ruby/integration_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'concurrent/future'
+
+require 'spec_helper'
+require 'ddtrace'
+
+RSpec.describe Datadog::Contrib::ConcurrentRuby::Integration do
+  let(:tracer) { ::Datadog::Tracer.new(writer: FauxWriter.new) }
+  let(:configuration_options) { { tracer: tracer } }
+
+  around do |example|
+    unmodified = ::Concurrent::Future.dup
+    Datadog.registry[:concurrent_ruby].patcher.instance_variable_set(:@done_once, {})
```

Comment:
I can see how you might want to redo patching when testing patching. However, does this need to be at the outermost level? Can this unpatching be done only in the `patching` context? Generally speaking we want to avoid hacking our patching like this, so limiting its footprint to only where its absolutely necessary would be best.

Reply:
Done

Reply:
I had to revert this change as all tests required the Future to be in known state as otherwise it would fail based on the order on execution